### PR TITLE
.circleci: Remove Python 2.7 tests

### PR DESCRIPTION
**Pull request by seemethere**
Python 2.7 was EOL as of January 1, 2020

PyTorch 1.4 was the last release to support Python 2.7

`libtorch` is currently named with a `2_7` suffix on the builds and should be changed to not represent any python version, but that can come in a later PR.

These are the list of tests that were ultimately removed:

<details>
<summary> Click me to expand </summary>

```
❯ git show | grep "^\-[ ]*name:"
-          name: pytorch_linux_xenial_py2_7_9_build
-          name: pytorch_linux_xenial_py2_7_9_test
-          name: pytorch_linux_xenial_py2_7_build
-          name: pytorch_linux_xenial_py2_7_test
-          name: binary_linux_manywheel_2_7mu_cpu_devtoolset7_build
-          name: binary_linux_conda_2_7_cpu_devtoolset7_build
-          name: binary_macos_conda_2_7_cpu_build
-          name: binary_linux_manywheel_2_7mu_cpu_devtoolset7_test
-          name: binary_linux_conda_2_7_cpu_devtoolset7_test
-          name: smoke_linux_manywheel_2_7m_cpu_devtoolset7_nightly
-          name: smoke_linux_manywheel_2_7mu_cpu_devtoolset7_nightly
-          name: smoke_linux_manywheel_2_7m_cu92_devtoolset7_nightly
-          name: smoke_linux_manywheel_2_7mu_cu92_devtoolset7_nightly
-          name: smoke_linux_manywheel_2_7m_cu100_devtoolset7_nightly
-          name: smoke_linux_manywheel_2_7mu_cu100_devtoolset7_nightly
-          name: smoke_linux_manywheel_2_7m_cu101_devtoolset7_nightly
-          name: smoke_linux_manywheel_2_7mu_cu101_devtoolset7_nightly
-          name: smoke_linux_conda_2_7_cpu_devtoolset7_nightly
-          name: smoke_linux_conda_2_7_cu92_devtoolset7_nightly
-          name: smoke_linux_conda_2_7_cu100_devtoolset7_nightly
-          name: smoke_linux_conda_2_7_cu101_devtoolset7_nightly
-          name: smoke_macos_wheel_2_7_cpu_nightly
-          name: smoke_macos_conda_2_7_cpu_nightly
-          name: binary_linux_manywheel_2_7m_cpu_devtoolset7_nightly_build
-          name: binary_linux_manywheel_2_7mu_cpu_devtoolset7_nightly_build
-          name: binary_linux_manywheel_2_7m_cu92_devtoolset7_nightly_build
-          name: binary_linux_manywheel_2_7mu_cu92_devtoolset7_nightly_build
-          name: binary_linux_manywheel_2_7m_cu100_devtoolset7_nightly_build
-          name: binary_linux_manywheel_2_7mu_cu100_devtoolset7_nightly_build
-          name: binary_linux_manywheel_2_7m_cu101_devtoolset7_nightly_build
-          name: binary_linux_manywheel_2_7mu_cu101_devtoolset7_nightly_build
-          name: binary_linux_conda_2_7_cpu_devtoolset7_nightly_build
-          name: binary_linux_conda_2_7_cu92_devtoolset7_nightly_build
-          name: binary_linux_conda_2_7_cu100_devtoolset7_nightly_build
-          name: binary_linux_conda_2_7_cu101_devtoolset7_nightly_build
-          name: binary_macos_wheel_2_7_cpu_nightly_build
-          name: binary_macos_conda_2_7_cpu_nightly_build
-          name: binary_linux_manywheel_2_7m_cpu_devtoolset7_nightly_test
-          name: binary_linux_manywheel_2_7mu_cpu_devtoolset7_nightly_test
-          name: binary_linux_manywheel_2_7m_cu92_devtoolset7_nightly_test
-          name: binary_linux_manywheel_2_7mu_cu92_devtoolset7_nightly_test
-          name: binary_linux_manywheel_2_7m_cu100_devtoolset7_nightly_test
-          name: binary_linux_manywheel_2_7mu_cu100_devtoolset7_nightly_test
-          name: binary_linux_manywheel_2_7m_cu101_devtoolset7_nightly_test
-          name: binary_linux_manywheel_2_7mu_cu101_devtoolset7_nightly_test
-          name: binary_linux_conda_2_7_cpu_devtoolset7_nightly_test
-          name: binary_linux_conda_2_7_cu92_devtoolset7_nightly_test
-          name: binary_linux_conda_2_7_cu100_devtoolset7_nightly_test
-          name: binary_linux_conda_2_7_cu101_devtoolset7_nightly_test
-          name: binary_linux_manywheel_2_7m_cpu_devtoolset7_nightly_upload
-          name: binary_linux_manywheel_2_7mu_cpu_devtoolset7_nightly_upload
-          name: binary_linux_manywheel_2_7m_cu92_devtoolset7_nightly_upload
-          name: binary_linux_manywheel_2_7mu_cu92_devtoolset7_nightly_upload
-          name: binary_linux_manywheel_2_7m_cu100_devtoolset7_nightly_upload
-          name: binary_linux_manywheel_2_7mu_cu100_devtoolset7_nightly_upload
-          name: binary_linux_manywheel_2_7m_cu101_devtoolset7_nightly_upload
-          name: binary_linux_manywheel_2_7mu_cu101_devtoolset7_nightly_upload
-          name: binary_linux_conda_2_7_cpu_devtoolset7_nightly_upload
-          name: binary_linux_conda_2_7_cu92_devtoolset7_nightly_upload
-          name: binary_linux_conda_2_7_cu100_devtoolset7_nightly_upload
-          name: binary_linux_conda_2_7_cu101_devtoolset7_nightly_upload
-          name: binary_macos_wheel_2_7_cpu_nightly_upload
-          name: binary_macos_conda_2_7_cpu_nightly_upload
-          name: "pytorch-linux-xenial-cuda9-cudnn7-py2"
-          name: "pytorch-linux-xenial-py2.7.9"
-          name: "pytorch-linux-xenial-py2.7"
-          name: binary_linux_manywheel_2_7mu_cpu_devtoolset7_build
-          name: binary_linux_conda_2_7_cpu_devtoolset7_build
-          name: binary_macos_conda_2_7_cpu_build
-          name: binary_linux_manywheel_2_7mu_cpu_devtoolset7_test
-          name: binary_linux_conda_2_7_cpu_devtoolset7_test
-          name: "pytorch-linux-xenial-cuda9-cudnn7-py2"
-          name: "pytorch-linux-xenial-py2.7.9"
-          name: "pytorch-linux-xenial-py2.7"
```
Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

</summary>

### CODE REVIEW

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit 6944a090:


* **1/1** failures introduced in this PR


---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

**Comment by seemethere** [comment]
I believe test failures are unrelated to the changes presented in this PR

File: .circleci/verbatim-sources/workflows-binary-builds-smoke-subset.yml [hunk 1]
```diff
@@ -46,24 +34,12 @@
...
-      - binary_mac_build:
-          name: binary_macos_conda_2_7_cpu_build
-          build_environment: "conda 2.7 cpu"
-          requires:
-            - setup
       - binary_mac_build:
           name: binary_macos_libtorch_2_7_cpu_build
           build_environment: "libtorch 2.7 cpu"
           requires:
             - setup
```
Comment (yns88) [inline .circleci/verbatim-sources/workflows-binary-builds-smoke-subset.yml]: Is this intentionally left behind?

**Review by ezyang — CHANGES_REQUESTED** [review]
This is a really good start.

However, we have to do the hard work of verifying that there isn't any testing that is hardcoded to Python 2, and would be lost if we dropped it this way.  Grepping for py2, I see the following special cases:

```
  if [[ "$BUILD_ENVIRONMENT" == *-xenial-cuda9-cudnn7-py2* ]]; then
    # TODO: move this to Docker
    sudo apt-get -qq update
    sudo apt-get -qq install --allow-downgrades --allow-change-held-packages openmpi-bin libopenmpi-dev
    sudo apt-get -qq install --no-install-recommends openssh-client openssh-server
    sudo mkdir -p /var/run/sshd
  fi
```

So we need to ensure that testing with MPI is still being covered somewhere else.

**Pull request closed without merging**